modules/packages/ArgumentParser.chpl

```diff
@@ -425,19 +507,69 @@ module ArgumentParser {
     }
  }
 
+  /*
+  A parser that performs the following functions:
+
+    1. Stores definitions of valid arguments/flags/options
+    2. Parses an array of arguments as passed from the command line
+       (where args[0] is typically the name of the executable)
+    3. Throws an ArgumentError if invalid or unrecognized arguments are encountered
+
+  The parser can process command line arguments that include or omit an equals
+  sign ``=``, with short ``-`` or long ``--`` indicators.
+
+  We can declare a new argumentParser like so:
+
+  .. code-block:: chapel
+
+    var parser = new argumentParser();
+
+  **Limitations**
+
+   * Cannot place short flags together. That is, if a parser has defined short
+     flags ``-c``, ``-a`` and ``-t``, we cannot handle ``-cat``.
+
+   * Short options must be separated from their values with an equals sign or
+     spaces. That is, ``-cValue`` is not supported and should be entered as either
+     ``-c Value`` or ``-c=Value``.
+
+   * Numeric types are not supported and will be passed as strings.
+
+   * Negative numbers on the command line may cause issues.
+
+   * Shadowing config vars or Chapel arguments can be accomplished by placing
```

Comment:
I'm not sure "Shadowing" will get the idea across.

Maybe "Chapel programs currently do config variable processing on the arguments. If you have arguments that conflict with config variables, you can place them after `--` on the command line in order to skip the config variable processing. If a passthrough is also specified as `--` then these arguments will also be captured by the passthrough. If that is not desired, you can create a different passthrough delimiter, e.g. `++`.